src/_pytest/assertion/truncate.py

```diff
@@ -12,16 +12,24 @@
 
 
 DEFAULT_MAX_LINES = 8
-DEFAULT_MAX_CHARS = 8 * 80
+DEFAULT_MAX_CHARS = DEFAULT_MAX_LINES * 80
 USAGE_MSG = "use '-vv' to show"
 
 
-def truncate_if_required(
-    explanation: list[str], item: Item, max_length: int | None = None
-) -> list[str]:
+def truncate_if_required(explanation: list[str], item: Item) -> list[str]:
     """Truncate this assertion explanation if the given test item is eligible."""
     if _should_truncate_item(item):
-        return _truncate_explanation(explanation)
+        max_lines = int(
+            item.config.getini("truncation_limit_lines") or DEFAULT_MAX_LINES
```

Comment:
Do we want to disable truncation altogether if users set `truncation_limit_lines` to `0`?

I believe so, in which case we need to handle `None` explicitly here, as `0` will short circuit to `DEFAULT_MAX_LINES` (same applies to the code below).

Reply:
I guess, that may be better to implement this logic as an additional check to `_should_truncate_item` method?

Reply:
Do we want to disable truncation if and only if both conditions (`truncation_limit_lines` and `truncation_limit_chars` are set to `0`) are met?

Reply:
Ahh yes, makes more sense to have this logic in `_should_truncate_item`.

Reply:
However, we still do need to have an explicit `0` handling, not here, but further down the stack (Here `0` is interpreted as a string and thus will be passed further into function, being transformed into `int` value), because in the current situation full truncations are possible by setting one of the values to `0`.

I think of a scenario as follows:
If either of `truncation_limit_lines` and `truncation_limit_chars` are set to `0` (but not both, that will be caught by `_should truncate_item`), we disable that exact truncation, but try to truncate with the other method, to avoid unwanted behaviour.

So, if the `truncation_limit_lines` is set to `0`, we do not try to truncate lines, but we try to truncate by `truncation_limit_chars`, and vice versa. 

That will look something like this
```diff
@@ -79,10 +88,11 @@ def _truncate_explanation(
     ):
         return input_lines
     # Truncate first to max_lines, and then truncate to max_chars if necessary
-    truncated_explanation = input_lines[:max_lines]
+    if max_lines > 0:
+        truncated_explanation = input_lines[:max_lines]
     truncated_char = True
     # We reevaluate the need to truncate chars following removal of some lines
-    if len("".join(truncated_explanation)) > tolerable_max_chars:
+    if len("".join(truncated_explanation)) > tolerable_max_chars and max_chars > 0:
```